docs/contrib/DeveloperGuide.md

```diff
@@ -318,6 +318,16 @@ clang-tidy -p ./build source_file.cpp
 clang-format -i -style=file:.clang-format source_file.cpp
 ```
 
+To run `clang-tidy` on all files, you can use the following script that runs
+`clang-tidy` in parallel and also caches the results, so subsequent runs
+only have to operate on changed files.
+
+```shell
+cmake . -B build  # generate build files
+ln -sf build/compile_commands.json .  # make compilation db visible
```

Comment:
Btw, it would be nice to eliminate the need for the link and just take it as an arg or envar